test/es6module/bug_OS17830745.js

```diff
@@ -0,0 +1,23 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// Resolving module2 will result in error (can't find the module file).
+// This causes us to mark module3 as having an error.
+// Later we resolve module0 which parses correctly.
+// This causes us to notify parents which will try to instantiate module3 because
+// module0 did not have an error. However, module3 has children modules which were 
+// not parsed. That put module3 into error state, we should skip instantiating
+// module3.
+
+WScript.RegisterModuleSource('module0_e0565b64-3435-42c1-ad1f-6376fb7af915.js', `
```

Comment:
>module0_e0565b64 [](start = 30, length = 16)

nit: this guid in the module name makes this hard to follow? could we remove this?

Reply:
Sure, these are just taken from an exprgen bug file. I'll cleanup the test case a big for readability.